src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -587,9 +587,10 @@ private void MakeExplicitParameterTypeInferences(Binder binder, BoundExpression
             {
                 ExplicitParameterTypeInference(argument, target, ref useSiteDiagnostics);
             }
-            else if (argument.Kind == BoundKind.TupleLiteral)
+            else if (argument.Kind == BoundKind.TupleLiteral &&
+                MakeExplicitParameterTypeInferences(binder, (BoundTupleLiteral)argument, target, isExactInference, ref useSiteDiagnostics))
             {
-                MakeExplicitParameterTypeInferences(binder, (BoundTupleLiteral)argument, target, isExactInference, ref useSiteDiagnostics);
+                ;
```

Comment:
>; [](start = 16, length = 1)

Please add a comment explaining the control-flow here (perhaps instead of the empty statement)

Reply:
@gafter 
>Please add a comment explaining the control-flow here (perhaps instead of the empty statement)

Would saying "We were able to do inference from elements of the tuple literal, no need to try to infer from literal's type" be sufficient?

Reply:
I'd just make it a sequence of separate if statements with returns.  Trying to preserve "else if" indeed makes it look strange.


---
In reply to: [140590732](https://github.com/dotnet/roslyn/pull/22295#discussion_r140590732) [](ancestors = 140590732)

Reply:
@AlekseyTs Yes, that would be a perfect comment to help the reader.